### PR TITLE
feature: add a data mapper to help mapping data from the resource data object to a property data

### PR DESCRIPTION
**Pull request by johnkrovitch**

### CODE REVIEW

File: src/Resource/Metadata/AbstractProperty.php [hunk 1]
```diff
@@ -7,7 +7,7 @@
 use LAG\AdminBundle\Validation\Constraint\TemplateValid;
 use Symfony\Component\Validator\Constraints as Assert;
 
-abstract class AbstractProperty implements PropertyInterface
+abstract class AbstractProperty implements \LAG\AdminBundle\Resource\Metadata\PropertyInterface
```
Comment (lutangar) [inline src/Resource/Metadata/AbstractProperty.php:10]: Could be a use statement instead.

File: src/Resource/DataMapper/DataMapperInterface.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+declare(strict_types=1);
+
+namespace LAG\AdminBundle\Resource\DataMapper;
+
+use LAG\AdminBundle\Resource\Metadata\PropertyInterface;
+
+/**
+ * Map resource data to a property data. Use the property path to retrieve the property data. The property path should
+ * follow the PropertyAccess syntax. A property path set to "true" returns the resource data. A property path set to
```
Comment (lutangar) [inline src/Resource/DataMapper/DataMapperInterface.php:11]: Not sure this contract is tested in `DataMapperTest`.

Comment (johnkrovitch) [reply src/Resource/DataMapper/DataMapperInterface.php:11]: This is tested in the two last items of the values provider :

```
        $data = new \stdClass();
        $data->id = 666;

        yield 'object' => [new Text(name: 'some', propertyPath: true), $data, $data];

        $data = new \stdClass();
        $data->id = 666;

        yield 'no_mapping' => [new Text(name: 'some', propertyPath: false), $data, null];
    }
```

File: src/Resource/Metadata/AbstractProperty.php (hunk 1 above)
Comment (johnkrovitch) [reply src/Resource/Metadata/AbstractProperty.php]: Fixed

**Pull request merged**